main.go

```diff
@@ -168,7 +170,18 @@ func main() {
 		if err != nil {
 			glog.Fatal(err)
 		}
-		if err = hook.Start(*webhookConfigName); err != nil {
+
+		jobNamespace, err := kubeClient.CoreV1().Namespaces().Get(*namespace, metav1.GetOptions{})
+		namespaceSelector := map[string]string{}
+		namespaceSelector["webhookNamespaceSelectorLabel"] = *namespace
```

Comment:
Add label `webhookNamespaceSelectorLabel: <sparkJobNamespace>`

Reply:
OK, this is not what I was suggesting. I don't think it's a good idea to modify the metadata of a user's namespace. What I was thinking is to check the namespace of each pod in the webhook and simply return a `AdmissionResponse` with `Allowed` set to true if `namespace` is set and  the pod is not in that namespace. This is not ideal, but works. Unfortunately, the `Webhook` API doesn't support selecting namespaces using their names, which to me seems the most natural way of selecting a single namespace.

Reply:
Another alternative is to get the `Namespace` object with the name specified through `--namespace` in `webhook.go`, and get the labels on the `Namespace`, and use them to construct the namespace selector. If the `Namespace` resource has no labels, fall back to the filtering mechanism I mentioned above.

Reply:
Yeah if only webhook API supports selecting a namespace using its name, that would be convenient. I'll come back to this PR and finish it after the holiday break.

Reply:
@liyinan926 I pushed a new commit using your suggested approach. Please take a look. Now no metadata is added/modified. The decision of whether to mutate a pod or not is made solely by comparing the pod's namespace and the pre-set job namespace.

Reply:
@liyinan926 Have you got a chance to review the changes?

Reply:
Sorry I haven't got a chance to review. I will try to get to this asap.